js/ui/hierarchical_collection/ui.data_adapter.js

```diff
@@ -321,8 +321,12 @@ const DataAdapter = Class.inherit({
     getNodeByItem: function(item) {
         let result = null;
 
+        const itemKey = this.options.dataAccessors.getters.key(item);
         each(this._dataStructure, function(_, node) {
-            if(node.internalFields.item === item) {
+            const isNodeFound = itemKey
+                ? node.internalFields.key === itemKey
```

Comment:
"itemKey ? node.internalFields.key === itemKey : node.internalFields.item === item" 

[null, undefined, false, 0] key values means that there is no key and "node.internalFields.item === item" should be used
Are you sure?

Reply:
Yep, fixed.
Now, I check the key only if it's value doesn't equals the undefined value